Added support for HealthCheckUnhealthyDurationBeforeReplacement under launchSpecification

sharadkesarwani opened the pull request:
Added support for HealthCheckUnhealthyDurationBeforeReplacement under launchSpecification.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-18145

sharadkesarwani committed:
Added support for HealthCheckUnhealthyDurationBeforeReplacement under launchSpecification

Added support for HealthCheckUnhealthyDurationBeforeReplacement under launchSpecification

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-17860-support-healthCheckUnhealthyDurationBeforeReplacement-ocean-aws

sharadkesarwani committed:
updated changelog

updated changelog

chandra1-n commented on CHANGELOG.md:
```diff
@@ -1,4 +1,7 @@
 ## Unreleased
+# 1.165.0 (March, 18 2024)
```
please modify the date

sharadkesarwani replied:
Updated latest date.

chandra1-n reviewed (changes requested):
please add this attribute in test

sharadkesarwani committed:
Updated changelog and added `health_check_unhealthy_duration_before_replacement ` attribute to testcase

Updated changelog and added `health_check_unhealthy_duration_before_replacement ` attribute to testcase

sharadkesarwani committed:
fixed typo and indentation

fixed typo and indentation
